### DISCUSSION TITLE
Request for a Stepper/Steps Component

### DISCUSSION THREAD

**Discussion by waqarthelive** [shadcn-ui/ui#6219] [Blocks Request]
### Description

We need a stepper component for our application that guides users through a multi-step process. Below are the requirements:
### Functionality:
1. Should support a two-step or multi-step process.
2. Users should be able to navigate between steps using Next and Back buttons.
3. Each step should display a unique content area with validation capabilities (if applicable).
4. Should handle both linear (steps completed in sequence) and non-linear (steps completed in any order) workflows.
5. Should display a progress indicator (e.g., numbers, titles, or icons) to show the current step.
### Design:
1. The component should match the application's design system (e.g., color scheme, typography).
2. Steps should be clearly marked and visually distinguishable (active step, completed step, pending step).
3. The layout should be responsive and work well on both desktop and mobile devices.

### Example

https://ant.design/components/steps

**Comment by resatyildiz** [comment]
As additional;
- it should be supported browser popstate event.

**Comment by resatyildiz** [comment]
> ### Description
> We need a stepper component for our application that guides users through a multi-step process. Below are the requirements:
> 
> ### Functionality:
> 1. Should support a two-step or multi-step process.
> 2. Users should be able to navigate between steps using Next and Back buttons.
> 3. Each step should display a unique content area with validation capabilities (if applicable).
> 4. Should handle both linear (steps completed in sequence) and non-linear (steps completed in any order) workflows.
> 5. Should display a progress indicator (e.g., numbers, titles, or icons) to show the current step.
> 
> ### Design:
> 1. The component should match the application's design system (e.g., color scheme, typography).
> 2. Steps should be clearly marked and visually distinguishable (active step, completed step, pending step).
> 3. The layout should be responsive and work well on both desktop and mobile devices.
> 
> ### Example
> https://ant.design/components/steps

Also ant design stepper good example for this component

**Comment by resatyildiz** [comment]
I found a PR: https://github.com/shadcn-ui/ui/pull/318
